[Request for help from TypeScript users] Re-enable `gulp typestest` in GitHub Actions

*PLEASE NOTE:* There's a risk that the type-definitions themselves are also broken now (and not just the tests), which means that unless this is fixed very soon the next PDF.js release may not have working types.

---

This was *temporary* disabled in PR #16698, and should be re-enabled as soon as possible.

0. If you're new to the project, please see https://github.com/mozilla/pdf.js#getting-the-code and https://github.com/mozilla/pdf.js/wiki/Contributing
1. Revert commit https://github.com/mozilla/pdf.js/commit/d9350c38997b31ee9f0058f438b332997eb53e07 locally, using `git revert d9350c3 --no-edit`.
3. Update the following code to handle (at least) the new `display-svg` import:
   https://github.com/mozilla/pdf.js/blob/43fc78899fea71cdadb4883aeeec51b849c29b3d/gulpfile.mjs#L1531-L1548
4. Run `gulp typestest`, and `gulp ci-test`, to validate your changes.
5. Finally, ensure that both `gulp jsdoc` and `gulp web` still work.

Seeing this issue as well. 
```
node_modules/pdfjs-dist/types/src/display/api.d.ts:1472:10 - error TS2304: Cannot find name 'SVGGraphics'.
1472 export { SVGGraphics };
```
Any workaround?

--- a/gulpfile.mjs
+++ b/gulpfile.mjs
@@ -1549,19 +1549,8 @@ gulp.task("jsdoc", function (done) {
 
 gulp.task("types", function (done) {
   console.log("### Generating TypeScript definitions using `tsc`");
-  const args = [
-    "target ESNext",
-    "allowJS",
-    "declaration",
-    `outDir ${TYPES_DIR}`,
-    "strict",
-    "esModuleInterop",
-    "forceConsistentCasingInFileNames",
-    "emitDeclarationOnly",
-    "moduleResolution node",
-  ].join(" --");
   exec(
-    `"node_modules/.bin/tsc" --${args} src/pdf.js web/pdf_viewer.component.js`,
+    `"node_modules/.bin/tsc" --outDir ${TYPES_DIR} --project .`,
     done
   );
 });

--- a/gulpfile.mjs
+++ b/gulpfile.mjs
@@ -266,6 +266,9 @@ function createWebpackConfig(
     viewerAlias["web-com"] = "web/chromecom.js";
     viewerAlias["web-print_service"] = "web/pdf_print_service.js";
   } else if (bundleDefines.GENERIC) {
+    // Aliases defined here must also be replicated in the paths section of
+    // the tsconfig.json file for the type generation to work.
+    // In the tsconfig.json files, the .js extension must be omitted.
     libraryAlias["display-fetch_stream"] = "src/display/fetch_stream.js";
     libraryAlias["display-l10n_utils"] = "web/l10n_utils.js";
     libraryAlias["display-network"] = "src/display/network.js";
@@ -1551,7 +1554,9 @@ gulp.task("types", function (done) {
   console.log("### Generating TypeScript definitions using `tsc`");
   exec(
     `"node_modules/.bin/tsc" --outDir ${TYPES_DIR} --project .`,
-    done
+    function () {
+      exec(`"node_modules/.bin/tsc-alias" --outDir ${TYPES_DIR}`, done);
+    }
   );
 });
 

--- a/gulpfile.mjs
+++ b/gulpfile.mjs
@@ -2469,6 +2469,7 @@ gulp.task(
   "ci-test",
   gulp.series(
     gulp.parallel("lint", "externaltest", "unittestcli"),
-    "lint-chromium"
+    "lint-chromium",
+    "typestest"
   )
 );

--- a/package.json
+++ b/package.json
@@ -50,6 +50,7 @@
     "stylelint-prettier": "^4.0.2",
     "terser": "^5.19.2",
     "through2": "^4.0.2",
+    "tsc-alias": "^1.8.7",
     "ttest": "^4.0.0",
     "typescript": "^5.1.6",
     "typogr": "^0.6.8",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -7,7 +7,16 @@
     "emitDeclarationOnly": true,
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
-    "moduleResolution": "node"
+    "moduleResolution": "node",
+    "paths": {
+      "display-fetch_stream": ["./src/display/fetch_stream"],
+      "display-l10n_utils": ["./web/l10n_utils"],
+      "display-network": ["./src/display/network"],
+      "display-node_stream": ["./src/display/node_stream"],
+      "display-node_utils": ["./src/display/node_utils"],
+      "display-svg": ["./src/display/svg"],
+      "pdfjs-lib": ["./src/pdf"]
+    }
   },
   "files": [
     "src/pdf.js",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "ESNext",
+    "allowJs": true,
+    "declaration": true,
+    "strict": true,
+    "emitDeclarationOnly": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "moduleResolution": "node"
+  },
+  "files": [
+    "src/pdf.js",
+    "web/pdf_viewer.component.js"
+  ]
+}

